akka-bbb-apps/src/main/scala/org/bigbluebutton/core/record/events/PluginEventPersistenceRecordEvent.scala

```diff
@@ -0,0 +1,28 @@
+package org.bigbluebutton.core.record.events
+
+import org.bigbluebutton.common2.util.JsonUtil
+
+class PluginEventPersistenceRecordEvent extends AbstractPluginRecordEvent {
+  import PluginEventPersistenceRecordEvent._
+
+  setEvent("PluginGeneratedEvent")
+
+  def setPersistedEventName(eventName: String) {
+    eventMap.put(PERSISTED_EVENT_NAME, eventName)
+  }
+
+  def setUserId(userId: String): Unit = {
+    eventMap.put(USER_ID, userId)
+  }
+
+  def setPayload(payload: Map[String, Object]) {
+    eventMap.put(PAYLOAD, JsonUtil.toJson(payload))
+  }
+}
+
+object PluginEventPersistenceRecordEvent {
+  protected final val USER_ID = "userId"
+  protected final val PERSISTED_EVENT_NAME = "persistedEventName"
```

Comment:
I think `eventName` is a better name (as suggested in the issue).

Reply:
You are right, it's not recommended to use the same name as it can be confusing.

what do you think about `pluginEventName`?